### PR TITLE
py-python-language-server: add new package

### PR DESCRIPTION
**Pull request by davydden**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-python-language-server/package.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class PyPythonLanguageServer(PythonPackage):
+    """An implementation of the Language Server Protocol for Python"""
+
+    homepage = "https://github.com/palantir/python-language-server"
+    url      = "https://github.com/palantir/python-language-server/archive/0.15.1.tar.gz"
+
+    version('0.15.1', 'deee6fe6776c4f59547939d356004650')
+
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-python-language-server/package.py]: Missing many required dependencies:
```python
    install_requires=[                                                          
        'configparser',                                                         
        'future>=0.14.0',                                                       
        'futures; python_version == "2.7"',                                     
        'jedi>=0.10',                                                           
        'json-rpc==1.10.8',                                                     
        'mccabe',                                                               
        'pycodestyle',                                                          
        'pydocstyle>=2.0.0',                                                    
        'pyflakes',                                                             
        'rope>=0.10.5',                                                         
        'yapf',                                                                 
        'pluggy'                                                                
    ],          
```

Comment (davydden) [reply var/spack/repos/builtin/packages/py-python-language-server/package.py]: thanks @adamjstewart , I added a few and fill in the rest later...

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-python-language-server/package.py]: They should all be `type=('build', 'run')`

**Comment by davydden** [comment]
@adamjstewart ping.

**Comment by adamjstewart** [comment]
@davydden what's up with the FIXME's? At least some of those packages already exist in Spack.

**Comment by davydden** [comment]
@adamjstewart 

> what's up with the FIXME's? At least some of those packages already exist in Spack.

by some you mean one? 😄 I fixed one FIXME.

**Comment by adamjstewart** [comment]
You'll need to add packages for the other dependencies before this can be merged.

**Comment by davydden** [comment]
> You'll need to add packages for the other dependencies before this can be merged.

fair enough. I guess I leave it as `up-for-grabs` due to the lack of time; also the latest release needs a bit different list of packages.

**Pull request closed without merging**